src/lib/Inference.hs

```diff
@@ -1309,9 +1404,35 @@ inferUVar = \case
   UMethodVar v -> do
     MethodBinding _ _ f <- lookupEnv v
     return f
-  UEffectVar _ -> throw NotImplementedErr "inferUVar::UEffectVar"
-  UEffectOpVar _ -> throw NotImplementedErr "inferUVar::UEffectOpVar"
-  UHandlerVar _ -> throw NotImplementedErr "inferUVar::UHandlerVar"
+  UEffectVar _ -> do
+    throw NotImplementedErr "inferUVar::UEffectVar"  -- TODO(alex): implement
+  UEffectOpVar v -> do
+    -- the following should never fail due to operation index since
+    -- users cannot name the "return" operation
+    EffectOpBinding (EffectOpDef effName (OpIdx i)) <- lookupEnv v
+    EffectDef effSourceName _ <- lookupEffectDef effName
+    let holeTy = HandlerDictTy (HandlerDictType effSourceName effName)
+    let holeCon = Con (HandlerHole (AlwaysEqual Nothing) holeTy)
```

Comment:
Ok, if you're not even preserving any useful source location, then we should definitely get rid of it (see my comment in Types/Core.hs)

Reply:
Handler holes are gone. #1071